app/src/main/java/org/p2p/wallet/history/repository/remote/TransactionDetailsRpcRepository.kt

```diff
@@ -1,40 +1,82 @@
+            this[RpcConstants.REQUEST_PARAMETER_KEY_COMMITMENT] =
+                RpcConstants.REQUEST_PARAMETER_VALUE_CONFIRMED
+        }
+        val requestsBatch = transactionSignature.map { signature ->
+            val params = listOf(signature.signature, encoding)
             RpcRequest(method = RpcConstants.REQUEST_METHOD_VALUE_GET_CONFIRMED_TRANSACTIONS, params = params)
         }
 
-        return rpcApi.getConfirmedTransactions(requestsBatch)
-            .map { it.result }
-            .let {
-                withContext(dispatchers.io) {
-                    transactionDetailsNetworkMapper.fromNetworkToDomain(it) { mintAddress ->
-                        userLocalRepository.findTokenData(mintAddress)?.mintAddress.orEmpty()
+        val transactions = rpcApi.getConfirmedTransactions(requestsBatch).map { it.result }
+
+        return fromNetworkToDomain(userPublicKey, transactions).onEach { transactionDetails ->
+            transactionDetails.status =
+                transactionSignature.first { it.signature == transactionDetails.signature }.status
+        }
+    }
+
+    private fun fromNetworkToDomain(
+        tokenPublicKey: String,
+        transactions: List<ConfirmedTransactionRootResponse>
+    ): List<TransactionDetails> {
+        val transactionDetails = mutableListOf<TransactionDetails>()
+        transactions.forEach { transaction ->
+            try {
+                val signature = transaction.transaction?.getTransactionId() ?: return@forEach
+                when {
+                    OrcaSwapInstructionParser.isTransactionContainsOrcaSwap(transaction) -> {
+                        val orcaSwapDetails =
+                            OrcaSwapInstructionParser.parse(signature = signature, transactionRoot = transaction)
+                        transactionDetails.add(orcaSwapDetails.getOrThrow())
+                    }
+                    SerumSwapInstructionParser.isTransactionContainsSerumSwap(transaction) -> {
+                        val serumSwapDetails =
+                            SerumSwapInstructionParser.parse(signature = signature, transactionRoot = transaction)
+                        transactionDetails.add(serumSwapDetails.getOrThrow())
+                    }
+                    else -> {
+                        transactionDetails.addAll(
+                            SolanaInstructionParser.parse(
+                                signature = signature,
+                                transactionRoot = transaction,
+                                userInteractor = userInteractor,
+                                userPublicKey = tokenPublicKey
+                            )
+                        )
                     }
                 }
+                transactionDetails.forEach {
+                    it.error = transaction.meta.error?.instructionError
+                }
+            } catch (e: Exception) {
+                Timber.e(e)
```

Comment:
should we skip parsing all the other transactions if only one has failed?

Reply:
removed this